### PR TITLE
[US22] Creating endpoints for posts

### PR DESCRIPTION
**Pull request by AndriiHorbenko**
Added entities for Post, PostLike, PostComment, PostShared, UserFollower, PostRepository, UserFollowRepository, PostRestContoller, UserFollowerRestController, PostService, UserFollowService, and DTOs: PostDtoResponse, PostDtoSave, PostLikeDto, UserDtoRequest and tests for PostDtoResponse, Post, PostRestController, PostService

### CODE REVIEW

File: social_network_api/src/main/java/com/danit/socialnetwork/dto/post/PostDtoResponse.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+@Data
+@NoArgsConstructor
+public class PostDtoResponse {
+
+  private Integer postId;
+  private String username;
+  private String name;
+  private String writtenText;
+  private byte[] photoFileByteArray;
+  //  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy:MM:dd HH:mm:ss")
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/dto/post/PostDtoResponse.java:24]: Please, remove comments.

File: social_network_api/src/main/java/com/danit/socialnetwork/model/PostComment.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  private String commentText;
+
+
+  @ManyToOne(targetEntity = DbUser.class)
+  @JoinColumn (name = "user_id")
+  private DbUser userPostComment;
+
+ /* @ManyToOne(targetEntity = Post.class)
+  @JoinColumn (name = "post_id")
+  private Post postPostComment;*/
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/model/PostComment.java:48]: The same

File: social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  @JoinColumn(name = "user_id")
+  private DbUser userPostLike;
+
+  @ManyToOne(targetEntity = Post.class)
+  @JoinColumn(name = "post_id")
+  private Post postPostLike;
+
+  public static PostLike from(PostLikeDto postLikeDto) {
+    return null;
+  }
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java:47]: Do we need this?

File: social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy:MM:dd HH:mm:ss")
+  private LocalDateTime createdDateTime;
+
+  @ManyToOne(targetEntity = DbUser.class)
+  @JoinColumn(name = "user_id")
+  private DbUser userPostLike;
+
+  @ManyToOne(targetEntity = Post.class)
+  @JoinColumn(name = "post_id")
+  private Post postPostLike;
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java:43]: postLike?

File: social_network_api/src/main/java/com/danit/socialnetwork/repository/PostRepository.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.List;
+
+public interface PostRepository extends JpaRepository<Post, Integer> {
+
+  List<Post> findAllByUserPost(DbUser user);
+
+  /*@Query(nativeQuery = true, value = "select * from POSTS p where p.user_id = : userID")
+  List<Post> findAllByUserId(Integer userId);*/
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/repository/PostRepository.java:15]: The same

File: social_network_api/src/main/java/com/danit/socialnetwork/rest/PostRestController.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.util.List;
+
+@Log4j2
+@RestController
+@RequiredArgsConstructor
+public class PostRestController {
+  private final PostService postService;
+
+  @GetMapping(path = "/home", produces = "application/json")
```
Comment (miegami) [inline social_network_api/src/main/java/com/danit/socialnetwork/rest/PostRestController.java:26]: Could we rename it to /posts?

File: social_network_api/src/main/resources/data.sql [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+insert into POSTS values (6, 'MTA6MjQ6MjY=', '2022-09-29 12:10:42.000000','Hello world 21', 2);
+insert into POSTS values (7, 'MTA6MjQ6MjY=', '2022-02-23 12:10:42.000000','Hello world 41', 4);
+
+insert into USER_FOLLOWS values (1,'true', 1, 2);
+insert into USER_FOLLOWS values (2, 'true', 1, 3);
+insert into USER_FOLLOWS values (3, 'true', 1, 4);
+insert into USER_FOLLOWS values (4, 'true', 2, 1);
+insert into USER_FOLLOWS values (5, 'true', 2, 4);
+insert into USER_FOLLOWS values (6, 'true', 3, 4);
+insert into USER_FOLLOWS values (7, 'true', 4, 1);
```
Comment (miegami) [inline social_network_api/src/main/resources/data.sql:31]: We don't need initial data.

File: social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java (hunk 4 above)
Comment (AndriiHorbenko) [reply social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java:43]: I have changed to postInPostLike

Comment (AndriiHorbenko) [reply social_network_api/src/main/java/com/danit/socialnetwork/model/PostLike.java:43]: To identify that it belongs field belongs to PostLike entity

File: social_network_api/src/main/resources/data.sql (hunk 7 above)
Comment (AndriiHorbenko) [reply social_network_api/src/main/resources/data.sql:31]: For testing yes, because we haven't yet created endpoints for following

Comment (AndriiHorbenko) [reply social_network_api/src/main/resources/data.sql:31]: @miegami I use this for testing!

**Review by miegami — APPROVED** [review]

**Review by Anastasiia2411 — APPROVED** [review]

**Pull request merged**